tests/components/broadlink/__init__.py

```diff
@@ -94,7 +104,7 @@ async def setup_entry(self, hass, mock_api=None, mock_entry=None):
         mock_entry.add_to_hass(hass)
 
         with patch(
-            "homeassistant.components.broadlink.device.blk.gendevice",
+            "homeassistant.components.broadlink.device.blk.device.__new__",
```

Comment:
I don't like this. Maybe I should change the library first to accept the class in gendevice(), optionally.